Abort scanning if there are more than 10000 files in a source directory.

petterreinholdtsen opened the pull request:
This make sure the service can not be overloaded by flodding one source
directory with files.  The number 10 000 is picked as such scan is fairly quick
while still being higher than what we commonly se in production.

Closes issue #109

dag-erling commented on libexec/scanner/scanner.c:
```diff
@@ -51,6 +51,8 @@
 #include <tsd/strutil.h>
 #include <tsd/percent.h>
 
+#define SCANLIMIT 10000
```
Please make this a variable rather than a macro so we can more easily make it configurable later. We might even want to move it into `struct scanpath`.

petterreinholdtsen commented:
I converted the limit to a variable and rewrote the code to store the number of processed files in the scanpath struct.  This allowed me to restructure the code closer to how it was before I added counting and time measurements (commit 1c231fc3f970eb47249b50c1f681d3d6fd9bae21).

petterreinholdtsen commented:
Is -m a good option for the scan limit?  The idea is to specify the maximum number of files returned.

dag-erling commented:
I guess `-m` is as good as any, but I would change the word used in the usage message to something that begins with an m (e.g. “maxfiles” or “maxscan”).

Another thing is that the `strtol()` code is incorrect, as it will accept negative numbers and numbers with trailing garbage. Here's an example of correct usage, taken from OpenPAM (although using `strotoul()` instead of `strtol()` is arguably a bug, as it will still accept negative numbers):

```
            n = strtoul(argv[0], &end, 10);
            if (end == argv[0] || *end != '\0' || n < 1 || n > 1000)
                    return (RET_USAGE);
```

I usually don't bother checking for `ERANGE`.

petterreinholdtsen commented:
I agree.  New and improved patch, this time with manual page updates as well.
